modulemd/v2/modulemd-yaml-util.c

```diff
@@ -555,15 +587,17 @@ modulemd_yaml_parse_document_type (yaml_parser_t *parser,
                   g_propagate_error (error, nested_error);
                   return FALSE;
                 }
+              /* TODO: Emit this scalar */
```

Comment:
Drop this TODO message.

Also, there's a bug in the original code here; if mdversion is 0, it might be due to a parse error *or* it might be that there was a literal zero in the document. We can't just return the `nested_error` as-is, since it might be NULL. Instead of `if (!mdversion)`, we should do `if (nested_error)`. The case where `mdversion == 0` is already handled at line 636.